File: AndroidStudioProjects/manseong2/app/src/main/java/com/example/manseong2/MainActivity.java
package com.example.manseong2;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private EditText edit1, edit2;
    private TextView result;
    private Button btnAdd, btnMin, btnMul, btnHal;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // UI 요소 연결
        edit1 = findViewById(R.id.edit1);
        edit2 = findViewById(R.id.edit2);
        result = findViewById(R.id.result);
        btnAdd = findViewById(R.id.btnAdd);
        btnMin = findViewById(R.id.btnMin);
        btnMul = findViewById(R.id.btnMul);
        btnHal = findViewById(R.id.btnHal);

        // 버튼 클릭 이벤트 추가
        btnAdd.setOnClickListener(v -> calculate('+'));
        btnMin.setOnClickListener(v -> calculate('-'));
        btnMul.setOnClickListener(v -> calculate('*'));
        btnHal.setOnClickListener(v -> calculate('/'));
    }

    // 사칙연산 계산 함수
    private void calculate(char operator) {
        String num1 = edit1.getText().toString().trim();
        String num2 = edit2.getText().toString().trim();

        // 입력값 확인
        if (num1.isEmpty() || num2.isEmpty()) {
            showToast("값을 입력하세요!");
            return;
        }

        double number1, number2;
        try {
            number1 = Double.parseDouble(num1);
            number2 = Double.parseDouble(num2);
        } catch (NumberFormatException e) {
            showToast("숫자를 입력하세요!");
            return;
        }

        double calcResult = 0;

        switch (operator) {
            case '+':
                calcResult = number1 + number2;
                break;
            case '-':
                calcResult = number1 - number2;
                break;
            case '*':
                calcResult = number1 * number2;
                break;
            case '/':
                if (number2 == 0) {
                    showToast("0으로 나눌 수 없습니다!");
                    return;
                }
                calcResult = number1 / number2;
                break;
        }

        result.setText("Result: " + calcResult);
    }

    // 토스트 메시지 표시 함수
    private void showToast(String message) {
        Toast.makeText(this, message, Toast.LENGTH_SHORT).show();
    }
}

File: AndroidStudioProjects/MyApplication/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.prolificinteractive.materialcalendarview.CalendarDay
import com.prolificinteractive.materialcalendarview.MaterialCalendarView


class MainActivity : AppCompatActivity() {

    private lateinit var calendarView: MaterialCalendarView
    private lateinit var addDiaryButton: Button
    private lateinit var addScheduleButton: Button
    private lateinit var scheduleTextView: TextView
    private var selectedDate: CalendarDay? = null
    private val schedulePrefs = "SchedulePrefs"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        calendarView = findViewById(R.id.MaterialCalendarView)
        addDiaryButton = findViewById(R.id.addDiaryButton)
        addScheduleButton = findViewById(R.id.addScheduleButton)
        scheduleTextView = findViewById(R.id.scheduleTextView)

        // 날짜 선택 리스너
        calendarView.setOnDateChangedListener { _, date, _ ->
            selectedDate = date
            displayScheduleForSelectedDate(date)
        }

        addDiaryButton.setOnClickListener {
            val intent = Intent(this, ViewDiaryActivity::class.java)
            intent.putExtra("selectedDate", selectedDate?.toString() ?: "")
            startActivity(intent)
        }

        addScheduleButton.setOnClickListener {
            // selectedDate가 null일 경우 기본 문자열을 전달하여 예외 방지
            val intent = Intent(this, ScheduleActivity::class.java)
            intent.putExtra("selectedDate", selectedDate?.let { "${it.year}-${it.month + 1}-${it.day}" } ?: "")
            startActivity(intent)
        }

        updateCalendarWithSchedules()
    }

    private fun displayScheduleForSelectedDate(date: CalendarDay) {
        val sharedPreferences = getSharedPreferences(schedulePrefs, MODE_PRIVATE)
        val dateKey = "${date.year}-${date.month + 1}-${date.day}"
        val schedule = sharedPreferences.getString("${dateKey}_schedule", "등록된 일정이 없습니다.")

        scheduleTextView.text = schedule
        scheduleTextView.setTextColor(Color.BLACK)
        scheduleTextView.visibility = View.VISIBLE
    }

    private fun updateCalendarWithSchedules() {
        val sharedPreferences = getSharedPreferences(schedulePrefs, MODE_PRIVATE)

        // 일정이 있는 날짜에 대한 마커 표시
        val markedDates = mutableSetOf<CalendarDay>()
        sharedPreferences.all.keys.forEach { key ->
            if (key.contains("_schedule")) {
                val parts = key.split("_")[0].split("-")
                if (parts.size == 3) {
                    val year = parts[0].toIntOrNull()
                    val month = parts[1].toIntOrNull()
                    val day = parts[2].toIntOrNull()
                    if (year != null && month != null && day != null) {
                        markedDates.add(CalendarDay.from(year, month - 1, day)) // month는 0-based
                    }
                }
            }
        }

        if (markedDates.isNotEmpty()) {
            calendarView.addDecorator(EventDecorator(Color.RED, markedDates))
        }
    }

    override fun onResume() {
        super.onResume()
        updateCalendarWithSchedules()
        selectedDate?.let { displayScheduleForSelectedDate(it) }
    }
}